libraries/rush-lib/src/api/PackageJsonEditor.ts

```diff
@@ -308,7 +308,10 @@ export class PackageJsonEditor {
     if (this._modified) {
       this._modified = false;
       this._sourceData = this._normalize(this._sourceData);
-      JsonFile.save(this._sourceData, this.filePath, { updateExistingFile: true });
+      JsonFile.save(this._sourceData, this.filePath, {
+        updateExistingFile: true,
+        jsonSyntax: JsonSyntax.Strict
+      });
```

Comment:
Note that strictly speaking there is minimal benefit to using the `JsonFile` API for saving `package.json` files over using `FileSystem.writeFile()` and `JSON.stringify()` (the latter is faster, in fact), when the output needs to be strict JSON. I guess this allows for cosmetic newlines?

Reply:
I think this will try to preserve the formatting, while JSON.stringify will reformat with the format that we specify? Although in the end it might not matter if the user has prettier configured (but that's just my assumption)

Reply:
Okay if I changed that block of code to this:

```js
      let stringified = JSON.stringify(this._sourceData, undefined, 2);
      stringified = Text.ensureTrailingNewline(stringified);
      FileSystem.writeFile(this.filePath, stringified);
```

When I tried this, one side effect of this is that when I run `rush add -p <pkg-name>`, it adds the dependencies alphabetically (i.e. actually fixing #2115 and maybe it's the more elegant solution to #2457)

What do you think @dmichon-msft  @chengcyber @iclanton ?

Reply:
@kenrick95 we've had the discussion about alphabetizing JSON files a few times in the past, and it's certainly something we would like to introduce at some point as an option (but not mandatory). I would consider it out of the scope of this PR though, and you can probably leave JSON file writing as-is for now.